doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -1724,9 +1724,55 @@ Ltac2 features a toplevel loop that can be used to evaluate expressions.
    together with its type. This command is pure in the sense that it does not
    modify the state of the proof, and in particular all side-effects are discarded.
 
+Compilation
+-----------
+
+Ltac2 definitions can be compiled to OCaml.  The compiled definitions
+will then run natively instead of using the interpreter.
+
+.. cmd:: Ltac2 Compile {* @qualid }
+
+   Compiles the named Ltac2 definitions and any other Ltac2 definitions they depend on
+   for use in the current Coq process.  The compiled code is used to execute the definitions
+   until the state reset (end of file, end of module, end of section).
```

Comment:
Dropping the compiled code at the end of a file, module or section seems very limiting.  Wouldn't it make it really hard for a user if they want to compile all their Ltac2 code?  Is this primarily about providing a way to clear out the cache of compiled code or something deeper?  It seems like it would be easy to end up recompiling the same code over and over.  If a compiled tactic calls a library, it could end compiling quite a lot of code (e.g. imagine they call an Ltac2 version of fiat-crypto)

Reply:
"Future work: cache compilation results across files" 1st post of the PR